core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/tracker/UserStoreManagerRegistry.java

```diff
@@ -111,4 +113,39 @@ public static Properties getUserStoreProperties(String className) {
         properties = getUserStoreManagers().get(className);
         return properties;
     }
+
+    /**
+     * Get if the user store is local or not.
+     *
+     * @param className User store name.
+     * @return boolean true if user store is a local.
+     */
+    public static boolean isLocalUserStore(String className) throws UserStoreException {
+
+        Object[] userStoreManagers = userStoreManagerTracker.getServices();
+        for (Object userStoreManagerObj : userStoreManagers) {
+            UserStoreManager userStoreManager = (UserStoreManager) userStoreManagerObj;
+            if (StringUtils.equals(className, userStoreManager.getClass().getName())) {
+                return userStoreManager.isLocalUserStore();
+            }
+        }
+        throw new UserStoreException(String.format("User store manager is not found for the given className: %s",
+                className));
+    }
+
+    /**
+     * Get all the available user store manager names and their type (local user store or not).
+     *
+     * @return Map<String, Boolean> Map of user store manager name and boolean indicating local or not.
+     */
+    public static Map<String, Boolean> getUserStoreManagersType() {
+
+        Map<String, Boolean> userStoreManagersType = new HashMap<>();
```

Comment:
Any specific reason to use the `Boolean` object type here? Can't we use primitive (`boolean`)?

Reply:
We cannot use primitive types to declare Map objects. Use of Map<String, boolean> map is not allowed in java.